metricproviders/datadog/datadog.go

```diff
@@ -129,7 +172,45 @@ func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alph
 	return measurement
 }
 
-func (p *Provider) parseResponse(metric v1alpha1.Metric, response *http.Response) (string, v1alpha1.AnalysisPhase, error) {
+func (p *Provider) createRequest(query string, apiVersion string, now int64, interval int64, url *url.URL) (*http.Request, error) {
+	if apiVersion == "v1" {
+		q := url.Query()
+		q.Set("query", query)
+		q.Set("from", strconv.FormatInt(now-interval, 10))
+		q.Set("to", strconv.FormatInt(now, 10))
+		url.RawQuery = q.Encode()
+
+		return &http.Request{Method: "GET"}, nil
+	} else if apiVersion == "v2" {
+		queryBody, _ := json.Marshal(datadogQuery{
```

Comment:
Can we handle this error in case user supplied query breaks marshaling.

Reply:
Commit: https://github.com/argoproj/argo-rollouts/pull/2592/commits/1e446dd612e3bffacde4ccb7eeab787c8cee4593